core/domain/blog_domain.py

```diff
@@ -0,0 +1,609 @@
+                    'Tag should not be empty, received: \'\'')
+
+            if not re.match(constants.TAG_REGEX, tag):
+                raise utils.ValidationError(
+                    'Tags should only contain lowercase letters and spaces, '
+                    'received: \'%s\'' % tag)
+
+            if (tag[0] not in string.ascii_lowercase or
+                    tag[-1] not in string.ascii_lowercase):
+                raise utils.ValidationError(
+                    'Tags should not start or end with whitespace, received: '
+                    '\'%s\'' % tag)
+
+            if re.search(r'\s\s+', tag):
+                raise utils.ValidationError(
+                    'Adjacent whitespace in tags should be collapsed, '
+                    'received: \'%s\'' % tag)
+
+        if strict:
+            if len(tags) == 0:
+                raise utils.ValidationError(
+                    'Atleast one tag should be selected')
+
+        if len(set(tags)) != len(tags):
+            raise utils.ValidationError(
+                'Some tags duplicate each other')
+
+    @classmethod
+    def require_valid_title(cls, title, strict):
+        """Checks whether the blog post title is a valid one.
+
+        Args:
+            title: str. The title to validate.
+            strict: bool. Enable strict checks on the blog post when the blog
+                post is published or is going to be published.
+        """
+
+        if not isinstance(title, python_utils.BASESTRING):
+            raise utils.ValidationError('Title should be a string.')
+
+        title_limit = constants.MAX_CHARS_IN_BLOG_POST_TITLE
+        if len(title) > title_limit:
+            raise utils.ValidationError(
+                'Blog Post title should at most have %d chars, received: %s'
+                % (title_limit, title))
+
+        if strict:
+            if len(title) == 0:
+                raise utils.ValidationError('Title should not be empty')
+            if not re.match(constants.VALID_BLOG_POST_TITLE_REGEX, title):
+                raise utils.ValidationError(
+                    'Title field contains invalid characters. Only words'
+                    '(a-zA-Z) separated by spaces are allowed. Received %s'
+                    % title)
+
+    @classmethod
+    def require_valid_url_fragment(cls, url_fragment):
+        """Checks whether the url fragment of the blog post is a valid one.
+
+        Args:
+            url_fragment: str. The url fragment to validate.
+        """
+
+        url_limit = constants.MAX_CHARS_IN_BLOG_POST_URL_FRAGMENT
+        utils.require_valid_url_fragment(
+            url_fragment, 'Blog Post URL Fragment', url_limit)
+
+    def to_dict(self):
+        """Returns a dict representing this blog post domain object.
+
+        Returns:
+            dict. A dict, mapping all fields of blog post instance.
+        """
+        return {
+            'id': self.id,
+            'author_name': user_services.get_user_id_from_username(
```

Comment:
using the service layer in the domain layer is prohibited, right?

/cc @vojtechjelinek

Reply:
I want to have the author's name in the frontend rather than the ID, Will you please suggest a better way to do this?
(I found similar examples in topic domain, to get the manager names in the topic rights dict.)

Reply:
@vojtechjelinek Can you please suggest whether it's fine to do it this way? (i.e, use services layer in domain class)

Reply:
We use that in some place but we would like to get rid of this behaviour.

Reply:
Thanks for the info @vojtechjelinek 

 @Rijuta-s In that case can we follow the pattern we do for exploration?

I.e, have author_id in domain layer but before sending the dict through API replace author_id with author_username [in controller layer]?